indexer-app/src/cmr/indexer/data/elasticsearch.clj

```diff
@@ -161,6 +161,24 @@
         (info "Existing index set:" (pr-str existing-index-set))
         (info "New index set:" (pr-str expected-index-set))))))
 
+(defn delete-index
+  "Delete an elastic index by name"
+  [context index]
+  (let [existing-index-set (index-set-es/get-index-set context idx-set/index-set-id)
+        expected-extra-granule-indexes (idx-set/index-set->reduced-extra-granule-indexes existing-index-set index)
+        expected-index-set (idx-set/index-set expected-extra-granule-indexes)
+        ]
+  (if (requires-update? existing-index-set expected-index-set)
+    (do
+      (info "Removing index to produce index set " (pr-str expected-index-set))
+      (index-set-svc/update-index-set context expected-index-set)
+      (info "Deleting collection index.")
+      (esi/delete-index (context->conn context) index))
+    (do
```

Comment:
This else branch was included to copy how the update-indexes function works, but am open to removing from both